src/XRP/issued-currency-client.ts

```diff
@@ -329,25 +331,108 @@ export default class IssuedCurrencyClient {
     amount: string,
     wallet: Wallet,
   ): Promise<TransactionResult> {
-    if (!XrpUtils.isValidXAddress(issuerXAddress)) {
+    return await this.sendTrustSetTransaction(
+      issuerXAddress,
+      currencyName,
+      amount,
+      undefined,
+      wallet,
+    )
+  }
+
+  /**
+   * Creates an authorized trust line between this XRPL account and another account.
+   * An authorized trust line will only be created if `requiredAuthorizedTrustlines` is called first.
+   *
+   * @see https://xrpl.org/authorized-trust-lines.html
+   *
+   * @param accountToAuthorize The X-Address of the address to authorize.
+   * @param currencyName The currency to authorize a trust line for.
+   * @param wallet The wallet creating the trust line.
+   */
+  public async authorizeTrustLine(
+    accountToAuthorize: string,
+    currencyName: string,
+    wallet: Wallet,
+  ): Promise<TransactionResult> {
+    // When authorizing a trust line, the value of the trust line is set to 0.
+    // See https://xrpl.org/authorized-trust-lines.html#authorizing-trust-lines
+    return await this.sendTrustSetTransaction(
+      accountToAuthorize,
+      currencyName,
+      '0',
+      TrustSetFlag.tfSetfAuth,
+      wallet,
+    )
+  }
+
+  /*
+   * Creates and sends a TrustSet transaction to be sent and executed on the XRPL.
+   *
+   * @param accountToTrust The account to extend trust to with a trust line.
+   * @param currencyName The name of the currency to create a trust line for.
+   * @param amount The maximum amount of debt to allow on this trust line.
+   * @param wallet A wallet associated with the account extending trust.
+   */
+  private async sendTrustSetTransaction(
+    accountToTrust: string,
+    currencyName: string,
+    amount: string,
+    flags: number | undefined,
+    wallet: Wallet,
+  ): Promise<TransactionResult> {
+    const trustSetTransaction = await this.prepareTrustSetTransaction(
+      accountToTrust,
+      currencyName,
+      amount,
+      flags,
+      wallet,
+    )
+
+    const transactionHash = await this.coreXrplClient.signAndSubmitTransaction(
+      trustSetTransaction,
+      wallet,
+    )
+
+    return await this.coreXrplClient.getFinalTransactionResultAsync(
+      transactionHash,
+      wallet,
+    )
```

Comment:
split this into 2 because `prepareTrustSetTransaction` has a lot of state changing going on